control-plane/tests/tests/pools.rs

```diff
@@ -0,0 +1,218 @@
+#![feature(allow_fail)]
+
+pub mod common;
+use common::*;
+
+#[actix_rt::test]
+async fn create_pool_malloc() {
+    let cluster = ClusterBuilder::builder().build().await.unwrap();
+    cluster
+        .rest_v0()
+        .create_pool(v0::CreatePool {
+            node: "mayastor".into(),
+            id: "pooloop".into(),
+            disks: vec!["malloc:///disk?size_mb=100".into()],
+        })
+        .await
+        .unwrap();
+}
+
+#[actix_rt::test]
+async fn create_pool_with_missing_disk() {
+    let cluster = ClusterBuilder::builder().build().await.unwrap();
+
+    cluster
+        .rest_v0()
+        .create_pool(v0::CreatePool {
+            node: "mayastor".into(),
+            id: "pooloop".into(),
+            disks: vec!["/dev/c/3po".into()],
+        })
+        .await
+        .expect_err("Device should not exist");
+}
+
+#[actix_rt::test]
+async fn create_pool_with_existing_disk() {
+    let cluster = ClusterBuilder::builder().build().await.unwrap();
+
+    cluster
+        .rest_v0()
+        .create_pool(v0::CreatePool {
+            node: "mayastor".into(),
+            id: "pooloop".into(),
+            disks: vec!["malloc:///disk?size_mb=100".into()],
+        })
+        .await
+        .unwrap();
+
+    cluster
+        .rest_v0()
+        .create_pool(v0::CreatePool {
+            node: "mayastor".into(),
+            id: "pooloop-new".into(),
+            disks: vec!["malloc:///disk?size_mb=100".into()],
+        })
+        .await
+        .expect_err("Disk should be used by another pool");
+
+    cluster
+        .rest_v0()
+        .destroy_pool(v0::DestroyPool {
+            node: "mayastor".into(),
+            id: "pooloop".into(),
+        })
+        .await
+        .unwrap();
+
+    cluster
+        .rest_v0()
+        .create_pool(v0::CreatePool {
+            node: "mayastor".into(),
+            id: "pooloop-new".into(),
+            disks: vec!["malloc:///disk?size_mb=100".into()],
+        })
+        .await
+        .expect("Should now be able to create the new pool");
+}
+
+#[actix_rt::test]
+async fn create_pool_again() {
+    let cluster = ClusterBuilder::builder().build().await.unwrap();
+
+    cluster
+        .rest_v0()
+        .create_pool(v0::CreatePool {
+            node: "mayastor".into(),
+            id: "pooloop".into(),
+            disks: vec!["malloc:///disk?size_mb=100".into()],
+        })
+        .await
+        .unwrap();
+
+    cluster
+        .rest_v0()
+        .create_pool(v0::CreatePool {
+            node: "mayastor".into(),
+            id: "pooloop".into(),
+            disks: vec!["malloc:///disk?size_mb=100".into()],
+        })
+        .await
+        .unwrap();
+}
+
+/// FIXME: CAS-710
```

Comment:
Maybe add to the ticket that the `#[allow_fail]` should be removed once fixed, so that someone less familiar with it knows what to do.